neo/Storage/Implementation/test/test_db_factory.py

```diff
@@ -0,0 +1,124 @@
+        self._db.write(b'00001.z', b'z')
+
+        self.assertEqual(self._db.get(b'00001.x'), b'x')
+        self.assertEqual(self._db.get(b'00001.y'), b'y')
+        self.assertEqual(self._db.get(b'00001.z'), b'z')
+
+    def test_delete_default(self):
+
+        self._db.write(b'00001.x', b'x')
+        self._db.delete(b'00001.z')
+
+        self.assertEqual(self._db.get(b'00001.z'), None)
+        self.assertEqual(self._db.get(b'00001.z', b'default'), b'default')
+
+    def test_iterator(self):
+
+        self._db.write(b'00001.x', b'x')
+        self._db.write(b'00001.y', b'y')
+        self._db.write(b'00001.z', b'z')
+
+        self._db.write(b'00002.w', b'w')
+        self._db.write(b'00002.x', b'x')
+        self._db.write(b'00002.y', b'y')
+        self._db.write(b'00002.z', b'z')
+
+        from neo.Storage.Interface.DBInterface import DBProperties
+
+        '''
+            Hhas to be converted as leveldb returns a custom iterator object, 
+            rocksdb just uses lists/dicts. Should not matter, still tests the 
+            same.
+        '''
+        def make_compatible(obj, to):
+            if not isinstance(obj, to):
+                new_obj = to(obj)
+                if isinstance(new_obj, dict):
+                    return new_obj.items()
+                return new_obj
+            return obj
+
+        with self._db.openIter(DBProperties(prefix=b'00001', include_value=True, include_key=False)) as iterator:
+
+            iterator = make_compatible(iterator, list)
+            self.assertEqual(len(iterator), 3)
+            self.assertIsInstance(iterator, list)
+
+        with self._db.openIter(DBProperties(prefix=b'00002', include_value=False, include_key=True)) as iterator:
+            iterator = make_compatible(iterator, list)
+            self.assertEqual(len(iterator), 4)
+            self.assertIsInstance(iterator, list)
+
+        with self._db.openIter(DBProperties(prefix=b'00002', include_value=True, include_key=True)) as iterator:
+            iterator = make_compatible(iterator, dict)
+            self.assertEqual(len(iterator), 4)
+            self.assertIsInstance(iterator, abc.ItemsView)
+
+        with self._db.openIter(DBProperties(prefix=None, include_value=True, include_key=True)) as iterator:
+            iterator = make_compatible(iterator, dict)
+            self.assertEqual(len(iterator), 7)
+            self.assertIsInstance(iterator, abc.ItemsView)
+
+        with self._db.openIter(DBProperties(prefix=None, include_value=False, include_key=True)) as iterator:
+            iterator = make_compatible(iterator, list)
+            self.assertEqual(len(iterator), 7)
+            self.assertIsInstance(iterator, list)
+
+        with self._db.openIter(DBProperties(prefix=None, include_value=True, include_key=False)) as iterator:
+            iterator = make_compatible(iterator, list)
+            self.assertEqual(len(iterator), 7)
+            self.assertIsInstance(iterator, list)
+
```

Comment:
what happens if we do  `DBProperties` with `include_value=False` and `include_key=False`? I know it doesn't make sense, but we should probably raise an exception as an invalid configuration. This can then be tested here as well

Reply:
Catching it now in the init and there is a test for it.